src/pages/ProfilesExplorerView/components/AnalyzeDiffFlameGraph.tsx

```diff
@@ -0,0 +1,71 @@
+import { css } from '@emotion/css';
+import { createAssistantContextItem, OpenAssistantButton, useAssistant } from '@grafana/assistant';
+import { Spinner } from '@grafana/ui';
+import React from 'react';
+
+import { buildPrompts } from './SceneAiPanel/domain/buildLlmPrompts';
+import { FetchParams, useFetchDotProfiles } from './SceneAiPanel/infrastructure/useFetchDotProfiles';
+
+interface AnalyzeDiffFlameGraphProps {
+  dataSourceUid: string;
+  profileMetricId: string;
+  isDiff: boolean;
+  fetchParams: FetchParams;
+}
+
+export function AnalyzeDiffFlameGraph({
+  dataSourceUid,
+  profileMetricId,
+  isDiff,
+  fetchParams,
+}: AnalyzeDiffFlameGraphProps) {
+  const { isAvailable } = useAssistant();
+
+  const { profileType, profiles, validationError, fetchError, isFetching } = useFetchDotProfiles(
+    isDiff,
+    fetchParams,
+    dataSourceUid,
+    profileMetricId
+  );
+
+  if (!isAvailable) {
+    return null;
+  }
+
+  const loading = validationError || fetchError || isFetching;
+
+  if (validationError || fetchError) {
+    return null;
+  }
+
+  if (!loading && profileType && profiles) {
```

Comment:
Can set loading to just `isFetching` or remove the const as `validationError` and `fetchError` checked below const.